src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,65 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String OPERATION_SUPPLY = "supply";
+    private static final String OPERATION_BUY = "buy";
+    private static final String DELIMITER = ",";
+    private static final int OPERATION_TYPE_INDEX = 0;
+    private static final int AMOUNT_INDEX = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String data = readFromFile(fromFileName);
+        String report = createReport(data);
+        writeReport(report, toFileName);
+    }
+
+    private String readFromFile(String fromFileName) {
+        StringBuilder data = new StringBuilder();
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                data.append(line).append(System.lineSeparator());
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("File " + fromFileName + " cannot be found");
+        }
+        return data.toString();
+    }
+
+    private String createReport(String data) {
+        int supplySum = 0;
+        int buySum = 0;
+        StringBuilder stringBuilder = new StringBuilder();
```

Comment:
better name suggestion. stringBuilder name tells nothing about this variable
```suggestion
        StringBuilder report= new StringBuilder();
```